lib/src/sha512.dart

```diff
@@ -0,0 +1,630 @@
+// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'digest.dart';
+import 'hash.dart';
+import 'hash_sink.dart';
+import 'utils.dart';
+
+/// How optimized do you want to be? Answer: yes.
+final bool _isJavascript = () {
```

Comment:
I heard from lrn that the more correct way to do this is `const bool.fromEnvironment('dart.library.js')`.

But, in this case we should probably do conditional imports instead. This also allows for code size reduction.
```dart
import 'sha512_fastsinks.dart'
  if (dart.library.js) 'sha512_slowsinks.dart';
```

Then have the two files define `Sha<...>Sink`.

Reply:
What Jonas says.

If we ever add `Uint64List` on a platform with web numbers, this test would fail. We *probably* won't, but you never know when browsers decide that memory is plentiful and their numbers should be 128 bit floats instead. Or they add arbitrary precision integers. Or something.

The *absolutely best* way would be having a declaration of `dart.integers.int64` or `dart.integers.web` depending on the platform, so you switch on the actual integer type, not a property which happens to (currently) be correlated to the integer type.
We don't have that yet. Until such time, I'd use `dart.library.js` as the best correlated property (it works even when compiling to node.js, which `dart.library.html` doesn't).

Reply:
Perfect - just what I was looking for.  I did a conditional import a long time ago and it was frowned upon.